src/core/util/clone.ts

```diff
@@ -54,6 +54,28 @@ export const cloneTree = <T extends Parentable<P>, P extends T>(
   return getFromPath(clonedRoot, pathToOldElement);
 };
 
+export const withCloneTree = <R, T1, T2>(
+  process: (clonedElement1: T1, clonedElement2: T2) => R,
+  [element1, element2, fallback]: [T1, T2, R]
```

Comment:
why use an array? why not single params?

Reply:
I used an array because I felt that this results in easier readable code when using this method, i.e. when calling it the most important part is specified first (the callback) with the (rather unimportant to the caller context) elements all grouped together, inspired by the React hooks functions. Of course it's not really the same as the semantics of the array elements are all the same in react hooks while here each position has a different semantic. It's obviously not as clear as I intended so I'm completely fine changing it.